velox/functions/lib/Re2Functions.cpp

```diff
@@ -496,46 +496,119 @@ bool matchExactPattern(
   return input.size() == 0;
 }
 
-std::pair<bool, int32_t> matchRelaxedFixedForwardAscii(
+namespace {
+class RelaxedPatternMatchResult {
+ public:
+  static RelaxedPatternMatchResult success(size_t cursor) {
+    return RelaxedPatternMatchResult{true, cursor, -1};
+  }
+
+  static RelaxedPatternMatchResult failed(
+      size_t cursor,
+      int32_t failingPatternIndex) {
+    return RelaxedPatternMatchResult{false, cursor, failingPatternIndex};
+  }
+
+  static RelaxedPatternMatchResult failedForTooShortInput() {
+    return RelaxedPatternMatchResult{false, 0, -1};
+  }
+
+  bool match() {
```

Comment:
- make this and other similar methods 'const'
- also make member variables 'const'